File: A3/src/dictionary/MyIDictionary.java
package dictionary;
import java.util.Collection;

import model.MyStmtException;

public interface MyIDictionary<T, U>
{
	void add(T v, U w);
	void delete(T v, U w);
	boolean isDefined(T v);
	void update(T v, U w);
	U lookup(T v) throws MyStmtException;
	boolean isDefinedV(U w);
	Collection<U> values();
	boolean isEmpty();
}

File: A7/Dictionary/MyIDictionary.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace A7.Dictionary
{
    interface MyIDictionary<T, U>
    {
        void Add(T v, U w);
        void Delete(T v, U w);
        Boolean IsDefinedU(T v);
        void Update(T v, U w);
        U Lookup(T v); //throws MyStmtException
        Boolean IsDefinedV(U w);
        ICollection<U> Values();
        Boolean IsEmpty();
    }
}
